src/aiida/plugins/utils.py

```diff
@@ -92,5 +93,6 @@ def get_version_info(self, plugin: str | type) -> dict[t.Any, dict[t.Any, t.Any]
             return self._cache[plugin]
 
         self._cache[plugin][KEY_VERSION_ROOT][KEY_VERSION_PLUGIN] = version_plugin
+        self._cache[plugin][KEY_VERSION_ROOT][KEY_VERSION_CACHE] = getattr(plugin, 'CACHE_VERSION', None)
```

Comment:
Not sure I follow the logic in this function, but should we try to only do this for Process subclasses?

Reply:
What other classes? The only other node subclasses are `Data` and we just went through why that would cause issues while it not even being necessary

Reply:
Sorry for not being clear. If I understand this function, it is defined for anything that a plugin defines, e.g. both Data and Process classes. Since now the CACHE_VERSION is only on Process classes, this means that `self._cache[plugin][KEY_VERSION_ROOT][KEY_VERSION_CACHE]` will always be `None` for Data classes? Or am I totally off mark here?

Reply:
The thing is that this `PluginVersionProvider` is only used by the `Process` class (through the `Runner`) to add the result to its attributes. So `Data` nodes are not affected.

Reply:
See this part of the code:

https://github.com/aiidateam/aiida-core/blob/69389e0387369d8437e1219487b88430b7b2e679/src/aiida/engine/processes/process.py#L713